### PR TITLE
Add documentation for new dbaas api

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

**Comment by ChiefMateStarbuck** [comment]
!deploy

**Review by ChiefMateStarbuck — CHANGES_REQUESTED** [review]
Failing lint

File: specification/resources/databases/models/pgbouncer.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    maxItems: 32
+  min_pool_size:
+    description: >-
+      Add more server connections to pool if below this number. Improves
+      behavior when usual load comes suddenly back after period of total
+      inactivity. The value is effectively capped at the pool size.
+    type: integer
+    minimum: 0
+    maximum: 10000
+    example: '0'
```
Comment (ChiefMateStarbuck) [inline specification/resources/databases/models/pgbouncer.yml:30]:
```suggestion
    example: 0
```

`'0'` is a string. removing the commas will make it an integer. There are a few places in this file you'll need to update this.

File: specification/resources/databases/models/mysql.yml [hunk 2]
```diff
@@ -1,6 +1,22 @@
 type: object
 
 properties:
+  backup_hour:
+    description: >-
+      The hour of day (in UTC) when backup for the service is started. New backup
+      is only started if previous backup has already completed.
```
Comment (yGabriel) [inline specification/resources/databases/models/mysql.yml:7]:
```suggestion
      The hour of day (in UTC) when backup for the service starts. New backup
      only starts if previous backup has already completed.
```

File: specification/resources/databases/models/mysql.yml [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+    minimum: 0
+    maximum: 23
+    type: integer
+    example: 3
+  backup_minute:
+    description: >-
+      The minute of an hour when backup for the service is started. New backup is
+      only started if previous backup has already completed.
```
Comment (yGabriel) [inline specification/resources/databases/models/mysql.yml:15]:
```suggestion
      The minute of the backup hour when backup for the service starts. New backup 
      only starts if previous backup has already completed.
```

File: specification/resources/databases/models/postgres.yml [hunk 4]
```diff
@@ -81,6 +81,22 @@ properties:
     minimum: -1
     maximum: 10000
     example: -1
+  backup_hour:
+    description: >-
+      The hour of day (in UTC) when backup for the service is started. New backup
+      is only started if previous backup has already completed.
```
Comment (yGabriel) [inline specification/resources/databases/models/postgres.yml:87]:
```suggestion
      The hour of day (in UTC) when backup for the service starts. New backup
     only starts if previous backup has already completed.
```

File: specification/resources/databases/models/postgres.yml [hunk 5]
```diff
@@ -81,6 +81,22 @@ properties:
...
+    minimum: 0
+    maximum: 23
+    type: integer
+    example: 3
+  backup_minute:
+    description: >-
+      The minute of an hour when backup for the service is started. New backup is
+      only started if previous backup has already completed.
```
Comment (yGabriel) [inline specification/resources/databases/models/postgres.yml:95]:
```suggestion
      The minute of the backup hour when backup for the service starts. New backup is
      only started if previous backup has already completed.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      Do not allow more than this many server connections per database
+      (regardless of user). Setting it to 0 means unlimited.
+    type: integer
+    minimum: 0
+    maximum: 2147483647
+    example: 0
+  autodb_idle_timeout:
+    description: >-
+      If the automatically created database pools have been unused this many
+      seconds, they are freed. If 0 then timeout is disabled. [seconds]
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:74]:
```suggestion
      If the automatically-created database pools have been unused this many
      seconds, they are freed. If 0, timeout is disabled.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      - session
+      - transaction
+      - statement
+    example: session
+    description: PGBouncer pool mode
+    type: string
+  autodb_max_db_connections:
+    description: >-
+      Do not allow more than this many server connections per database
+      (regardless of user). Setting it to 0 means unlimited.
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:66]:
```suggestion
      Only allows a maximum this many server connections per database
      (regardless of user). If 0, allows unlimited connections.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      If a server connection has been idle more than this many seconds it will
+      be dropped. If 0 then timeout is disabled. [seconds]
+    type: integer
+    minimum: 0
+    maximum: 86400
+    example: 600
+  autodb_pool_size:
+    description: >-
+      If non-zero then create automatically a pool of that size per user when
+      a pool doesn't exist.
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:50]:
```suggestion
      If non-zero, automatically creates a pool of that size per user when
      a pool doesn't exist.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      The pooler will close an unused server connection that has been
+      connected longer than this. [seconds]
+    type: integer
+    minimum: 60
+    maximum: 86400
+    example: 3600
+  server_idle_timeout:
+    description: >-
+      If a server connection has been idle more than this many seconds it will
+      be dropped. If 0 then timeout is disabled. [seconds]
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:42]:
```suggestion
      Drops server connections if they have been idle more than this many seconds. 
      If 0, timeout is disabled. 
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    maxItems: 32
+  min_pool_size:
+    description: >-
+      Add more server connections to pool if below this number. Improves
+      behavior when usual load comes suddenly back after period of total
+      inactivity. The value is effectively capped at the pool size.
+    type: integer
+    minimum: 0
+    maximum: 10000
+    example: 0
```
Comment (ChiefMateStarbuck) [inline specification/resources/databases/models/pgbouncer.yml:30]:
```suggestion
    example: 1
```

`0` is not a valid example. This is an odd bug with openAPI

File: specification/resources/databases/models/pgbouncer.yml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+  server_lifetime:
+    description: >-
+      The pooler will close an unused server connection that has been
+      connected longer than this. [seconds]
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:34]:
```suggestion
      The pooler closes any unused server connection that has been
      connected longer than this amount of seconds.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      type: string
+      enum:
+        - extra_float_digits
+        - search_path
+    maxItems: 32
+  min_pool_size:
+    description: >-
+      Add more server connections to pool if below this number. Improves
+      behavior when usual load comes suddenly back after period of total
+      inactivity. The value is effectively capped at the pool size.
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:26]:
```suggestion
      If current server connections are below this number, adds more. Improves
      behavior when usual load comes suddenly back after period of total
      inactivity. The value is effectively capped at the pool size.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    type: boolean
+    example: false
+  ignore_startup_parameters:
+    description: List of parameters to ignore when given in startup packet
+    type: array
+    example:
+      - extra_float_digits
+      - search_path
+    items:
+      description: Enum of parameters to ignore when given in startup packet
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:16]:
```suggestion
      description: Enum of parameters to ignore when given in startup packet.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 14]
```diff
@@ -0,0 +1,78 @@
+type: object
+description: PGBouncer connection pooling settings
+
+properties:
+  server_reset_query_always:
+    description: Run server_reset_query (DISCARD ALL) in all pooling modes
+    type: boolean
+    example: false
+  ignore_startup_parameters:
+    description: List of parameters to ignore when given in startup packet
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:10]:
```suggestion
    description: List of parameters to ignore when given in startup packet.
```

File: specification/resources/databases/models/pgbouncer.yml [hunk 15]
```diff
@@ -0,0 +1,78 @@
+type: object
+description: PGBouncer connection pooling settings
+
+properties:
+  server_reset_query_always:
+    description: Run server_reset_query (DISCARD ALL) in all pooling modes
```
Comment (yGabriel) [inline specification/resources/databases/models/pgbouncer.yml:6]:
```suggestion
    description: Run server_reset_query (DISCARD ALL) in all pooling modes.
```

**Review by ChiefMateStarbuck — APPROVED** [review]

**Pull request merged**